### PR TITLE
feat: ensure domain uniqueness

### PR DESCRIPTION
**Pull request by wa0x6e**
Toward https://github.com/snapshot-labs/workflow/issues/142
Together with https://github.com/snapshot-labs/snapshot-hub/pull/928

When saving space settings with domain, this PR will also save the domain in the `domain` column, and ensure that the domain is unique

See the hub PR for database update

### CODE REVIEW

File: src/helpers/actions.ts [hunk 1]
```diff
@@ -7,19 +7,21 @@ export async function addOrUpdateSpace(space: string, settings: any) {
...
 
   await db.queryAsync(query, [
     {
       id: space,
       name: settings.name,
       created: ts,
       updated: ts,
-      settings: JSON.stringify(settings)
+      settings: JSON.stringify(settings),
+      domain: settings.domain || null
```
Comment (ChaituVR) [inline src/helpers/actions.ts:19]: i think better to be undefined 
```suggestion
      domain: settings.domain || undefined
```

File: src/writer/settings.ts [hunk 2]
```diff
@@ -82,6 +82,17 @@ export async function verify(body): Promise<any> {
   const isAdmin = admins.includes(body.address.toLowerCase());
   const newAdmins = (msg.payload.admins || []).map(admin => admin.toLowerCase());
 
+  const anotherSpaceWithDomain = (
+    await db.queryAsync('SELECT 1 FROM spaces WHERE LOWER(domain) = LOWER(?) AND id != ? LIMIT 1', [
+      msg.payload.domain,
+      msg.space
+    ])
+  )[0];
```
Comment (ChaituVR) [inline src/writer/settings.ts:90]: We should restrict it to be always lowercase here https://github.com/snapshot-labs/snapshot.js/blob/master/src/schemas/space.json#L104-L107

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/helpers/actions.ts (hunk 1 above)
Comment (wa0x6e) [reply src/helpers/actions.ts:19]: What will it do on the SQL side when passing undefined, and when there is previous value ? Will it just skip this column, or set undefined ? 

This column is only accepting either NULL or string, so what will undefined be ?

Comment (ChaituVR) [reply src/helpers/actions.ts:19]: ahh i was assuming it to be `domain` inside JSON 😆

File: src/writer/settings.ts (hunk 2 above)
Comment (wa0x6e) [reply src/writer/settings.ts:90]: See https://github.com/snapshot-labs/snapshot.js/pull/1060

**Comment by ChaituVR** [comment]
Any idea about the test @wa0x6e

**Comment by wa0x6e** [comment]
> Any idea about the test @wa0x6e

was unable to fix the tests, they're failing because of other tests, since the database is shared among all integrations, which are run in parallel

Running the test individually work (use `dotenv -e test/.env.test yarn jest test/integration/writer/settings.test.ts`)

This test is disabled for now, so we can move on

**Review by ChaituVR — APPROVED** [review]
tAck

**Pull request merged**